app/forms/anti-affinity-group-member-add.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { useForm } from 'react-hook-form'
+
+import {
+  queryClient,
+  useApiMutation,
+  type AntiAffinityGroupMemberInstanceAddPathParams,
+  type Instance,
+} from '~/api'
+import { ComboboxField } from '~/components/form/fields/ComboboxField'
+import { HL } from '~/components/HL'
+import { useAntiAffinityGroupSelector } from '~/hooks/use-params'
+import { addToast } from '~/stores/toast'
+import { toComboboxItems } from '~/ui/lib/Combobox'
+import { Modal } from '~/ui/lib/Modal'
+
+export default function AddAntiAffinityGroupMemberForm({
+  availableInstances,
+  isModalOpen,
+  setIsModalOpen,
+}: {
+  availableInstances: Instance[]
+  isModalOpen: boolean
+  setIsModalOpen: (open: boolean) => void
+}) {
+  const { project, antiAffinityGroup } = useAntiAffinityGroupSelector()
+
+  const form = useForm<AntiAffinityGroupMemberInstanceAddPathParams>({
+    defaultValues: {
+      instance: '',
+    },
+  })
+
+  const onDismiss = () => {
+    setIsModalOpen(false)
+    form.reset()
+  }
+
+  const { mutateAsync: addMember } = useApiMutation('antiAffinityGroupMemberInstanceAdd', {
+    onSuccess(_data, variables) {
+      onDismiss()
+      queryClient.invalidateEndpoint('antiAffinityGroupMemberList')
+      queryClient.invalidateEndpoint('antiAffinityGroupView')
+      addToast(<>Member <HL>{variables.path.instance}</HL> added to anti-affinity group <HL>{antiAffinityGroup}</HL></>) // prettier-ignore
+    },
+  })
+
+  return (
+    <Modal isOpen={isModalOpen} onDismiss={onDismiss} title="Add instance to group">
+      <Modal.Body>
+        <Modal.Section>
+          <p className="text-sm text-gray-500">
+            Select an instance to add to the anti-affinity group{' '}
+            <HL>{antiAffinityGroup}</HL>.
+          </p>
+          <ComboboxField
+            label="Instance"
+            placeholder="Select an instance"
+            name="instance"
+            items={toComboboxItems(availableInstances)}
+            required
+            control={form.control}
+          />
+        </Modal.Section>
+      </Modal.Body>
+      <Modal.Footer
+        onDismiss={onDismiss}
+        onAction={() =>
+          addMember({
+            path: {
+              antiAffinityGroup,
+              instance: form.getValues('instance'),
+            },
+            query: { project },
+          })
+        }
+        actionText="Add to group"
+      />
+    </Modal>
```

Comment:
Using a `<form>` and going through the form hook's `handleSubmit` is better because it makes stuff work like pressing enter in the combobox to submit the modal. Example:

https://github.com/oxidecomputer/console/blob/ab06a337f81df76cd5fcf619089e3bd4d055270c/app/pages/SiloImagesPage.tsx#L162-L205

Reply:
As we figured out in chat, `Combobox` was explicitly eating the enter keypress when the combobox options are closed (precisely the situation we care about here). In 37a673383c7d1de68f13429432ca08b93c3472df I tweaked the logic in `Combobox` to only do that when `onEnter` is passed, i.e., only when we're using the combobox in a subform (only example currently is firewall rule form) where we want the subform to submit on enter rather than the whole form.